src/lib/Parser.hs

```diff
@@ -237,20 +239,20 @@ topDecl = dataDef <|> topLet
 topLet :: Parser UDecl
 topLet = do
   lAnn <- (char '@' >> letAnnStr <* (void eol <|> sc)) <|> return PlainLet
-  ~(ULet _ (p, ann) rhs, pos) <- withPos decl
+  ~(ULet _ (p, ann) rhs, pos) <- withPos $ decl `fallBackTo` interfaceInstance
```

Comment:
Let's avoid the fallback and just make `instance` a keyword. Just put it in `sourceBlock'`, similarly to how you deal with `interfaceDef`